amd/src/acceptmodal.js

```diff
@@ -0,0 +1,222 @@
+
+
+/**
+ * Add a create new group modal to the page.
+ *
+ * @module     tool_policy/acceptmodal
+ * @class      AcceptOnBehalf
+ * @package    tool_policy
+ * @copyright  2018 Marina Glancy
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+define(['jquery', 'core/str', 'core/modal_factory', 'core/modal_events', 'core/fragment', 'core/ajax', 'core/yui'],
+    function($, Str, ModalFactory, ModalEvents, Fragment, Ajax, Y) {
+
+        /**
+         * Constructor
+         *
+         * @param {int} contextid
+         *
+         * Each call to init gets it's own instance of this class.
+         */
+        var AcceptOnBehalf = function(contextid) {
+            this.contextid = contextid;
+
+            var stringKeys = [
+                {
+                    key: 'consentdetails',
+                    component: 'tool_policy'
+                },
+                {
+                    key: 'iagreetothepolicy',
```

Comment:
Should the modal use "I agree ..." given that the aggrement is made on someone else's behalf? What about having a new string like "Mark policies as agreed" or "Record agreement with policies" or so.

Reply:
this was on the mockups, the string can be changed later to whatever

Reply:
https://projects.invisionapp.com/share/WJFKUJA6BVD#/screens/279745958

Reply:
Ah, ok then.